GiniVision/Classes/Networking/DocumentServiceProtocol.swift

```diff
@@ -0,0 +1,187 @@
+                        fileName: String,
+                        docType: String = "",
+                        cancellationToken: BFCancellationToken? = nil,
+                        completion: @escaping UploadDocumentCompletion) {
+        giniSDK.sessionManager
+            .getSession()
+            .continueWith(block: sessionBlock(cancellationToken: cancellationToken))
+            .continueOnSuccessWith(block: { [weak self] _ in
+                return self?.giniSDK.documentTaskManager.createPartialDocument(withFilename: fileName,
+                                                                               from: document.data,
+                                                                               docType: docType,
+                                                                               cancellationToken: cancellationToken)
+            }).continueWith(block: { task in
+                if let createdDocument = task.result as? GINIDocument {
+                    print("📄 Created document with id:", createdDocument.documentId ?? "",
+                          "for vision document:", document.id)
+                    completion(.success(createdDocument))
+                } else if task.isCancelled {
+                    completion(.failure(AnalysisError.cancelled))
+                } else {
+                    completion(.failure(AnalysisError.documentCreation))
+                }
+                
+                return nil
+            })
+    }
+    
+    func deleteCompositeDocument(withId id: String) {
+        giniSDK.sessionManager
+            .getSession()
+            .continueWith(block: sessionBlock(cancellationToken: nil))
+            .continueOnSuccessWith(block: { [weak self] _ in
+                    self?.giniSDK.documentTaskManager.deleteCompositeDocument(withId: id,
+                                                                              cancellationToken: nil)
+            })
+    }
+    
+    func deletePartialDocument(withId id: String) {
+        giniSDK.sessionManager
+            .getSession()
+            .continueWith(block: sessionBlock(cancellationToken: nil))
+            .continueOnSuccessWith(block: { [weak self] _ in
+                self?.giniSDK.documentTaskManager.deletePartialDocument(withId: id,
+                                                                        cancellationToken: nil)
+            })
+    }
+    
+    func fetchExtractions(for documents: [PartialDocumentInfo],
+                          completion: @escaping AnalysisCompletion) {
+        let partialDocumentsInfo = documents.map { $0.toDictionary() }
+        analysisCancellationToken = BFCancellationTokenSource()
+        
+        giniSDK
+            .documentTaskManager
+            .createCompositeDocument(withPartialDocumentsInfo: partialDocumentsInfo,
+                                     fileName: "",
+                                     docType: "",
```

Comment:
Why are the file name and doc type always empty?

Reply:
For testing purposes. But I can add a default fileName,

Reply:
Actually in the next PR I added a "FileName" because it was failing when testing.

Reply:
Hehe, nice!